mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,131 @@
-  override def fullLoad(): Seq[(Long, Schema)] = ???
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Await
+import scala.util.{Failure, Try}
 
-  override def insert(schemas: Seq[(Long, Schema)]): Unit = ???
+class MongoConnector(mongoClient: MongoClient, mongoConfig: BaseMongoConfig) extends Connector with Logging {
 
-  override def createTable(): Unit = ???
+  private def parser: Parser = new Parser()
 
-  override def tableExists(): Boolean = ???
+  override def fullLoad(): Seq[(Long, Schema)] = {
 
-  override def tableCreationHint(): String = ???
+    log.debug(s"loading all schemas from collection ${mongoConfig.collection}")
+    val collection =
+      mongoClient
+        .getDatabase(mongoConfig.database)
+        .getCollection(mongoConfig.collection)
 
-  override def findSchema(id: Long): Option[Schema] = ???
+    val schemas: Seq[Try[(Long, Schema)]] =
+      Await.result(
+        collection.find().map { document =>
+          for {
+            key <- extract(document, "_id", _.asInt64().getValue)
+            schemaStr <- extract(document, "schema", _.asString().getValue)
+            schema <- Try(parser.parse(schemaStr))
+          } yield key -> schema
+        }.toFuture(),
+        mongoConfig.timeout
+      )
+    log.debug(s"${schemas.size} loaded from MongoDB")
+    // this way the first exception is thrown, but we can change this line to support different error handling strategies
+    schemas.map(_.get)
+  }
+
+  private def extract[A](d: Document, fieldName: String, f: BsonValue => A): Try[A] = {
+    d.filterKeys(k => k == fieldName)
+      .headOption
+      .fold[Try[A]](Failure(new RuntimeException(s"Cannot find $fieldName field in document"))) {
+        case (_, value) =>
+          Try(f(value)).recoverWith {
+            case t: Throwable =>
+              Failure(new RuntimeException(s"$fieldName was not of expected type", t))
+          }
+      }
+  }
+
+  override def insert(schemas: Seq[(Long, Schema)]): Unit = {
+
+    log.debug(s"inclusion of new schemas in the collection ${mongoConfig.collection}")
+
+    schemas.foreach {
+      case (id, schema) =>
+        val document = new BsonDocument
+        document.put("_id", bson.BsonInt64(id))
+        document.put("schema", bson.BsonString(schema.toString))
+        document.put("name", bson.BsonString(schema.getName))
+        document.put("namespace", bson.BsonString(schema.getNamespace))
+
+        insertIfNotExists(mongoClient.getDatabase(mongoConfig.database).getCollection(mongoConfig.collection), document)
+    }
+  }
+
+  private def insertIfNotExists(collection: MongoCollection[Document], document: BsonDocument): Unit = {
+    try {
+      Await.result(collection.insertOne(document).toFuture(), mongoConfig.timeout)
+    } catch {
+      case ex: MongoWriteException if ex.getError.getCategory == ErrorCategory.DUPLICATE_KEY =>
+        log.info("document already present, doing nothing")
+    }
+    Unit
```

Comment:
The instance of `Unit` is `()` :)
```suggestion
    ()
```